apps/src/templates/SchoolInfoConfirmationDialog.jsx

```diff
@@ -16,41 +16,88 @@ class SchoolInfoConfirmationDialog extends Component {
     schoolName: PropTypes.string,
     onUpdate: PropTypes.func,
     onConfirm: PropTypes.func,
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.object,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
       isOpen: true
     };
   }
 
-  handleUpdateClick = () => {
+  static defaultProps = {
+    schoolName: '',
+    onUpdate: () => {},
+    onConfirm: () => {},
+    scriptData: {},
+    onClose: () => {},
+    isOpen: true
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
+        .then(response => response.json())
+        .then(data => {
+          this.setState({
+            schoolName: data.school_name
+          });
+        })
+        .catch(error => this.setState({error}));
+    }
+  }
+
+  handleClickYes = () => {
+    fetch(
+      `/dashboardapi/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }`
+    )
+      .then(() => this.props.onClose())
+      .catch(error => this.setState({error}));
+  };
+
+  handleClickUpdate = () => {
     this.setState({showSchoolInterstitial: true});
   };
 
+  handleClickSave = () => {
+    this.props.onUpdate();
+
+    fetch('/dashboardapi/v1/users/me/update_user_info', {
+      method: 'PUT'
+    }).then(() => {
+      fetch('/dashboardapi/v1/user_school_infos/me').then(
+        () => this.props.onClose
+      );
+    });
+  };
+
   renderInitialContent() {
-    const {onConfirm} = this.props;
+    const {schoolName} = this.state;
     return (
       <Body>
         <div>
-          <p>{i18n.schoolInfoDialogDescription()} Lincoln Elementary School?</p>
+          <p>
+            {i18n.schoolInfoDialogDescription()} {schoolName}
```

Comment:
this will break for RTL languages, so i think we should update the `schoolInfoDialogDescription` string to include our `schoolName`:
```
schoolInfoDialogDescription: "Are you still at {schoolName}?",
```
then we can call it like this: `i18n. schoolInfoDialogDescription({schoolName})`